File: exercism/python/grade-school/grade_school.py
from collections import defaultdict
from sortedcontainers import SortedSet


class School(object):

    def __init__(self, name):
        self._name = name
        self._roster = defaultdict(SortedSet)

    def add(self, name, grade):
        self._roster[grade].add(name)

    def grade(self, grade):
        return tuple(self._roster[grade])

    def sort(self):
        return [(i, tuple(v)) for i, v in self._roster.items()]

File: exercism/cpp/grade-school/grade_school.cpp
#include "grade_school.h"

using grade_school::school;
using grade_school::GradeT;
using grade_school::RosterT;

void school::add(const string& name, int grade)
{
  auto &g = m_roster[grade]; // Didn't know this acts as a default
  g.push_back(name);
  std::sort(g.begin(), g.end());
}

GradeT school::grade(int grade)
{
  if (!m_roster.count(grade))
    return {}; // Eww
  return m_roster.at(grade);
}

